src/umpire/Umpire.cpp

```diff
@@ -168,4 +170,56 @@ std::vector<util::AllocationRecord> get_leaked_allocations(Allocator allocator)
   return get_allocator_records(allocator);
 }
 
+umpire::MemoryResourceTraits get_default_resource_traits(const std::string name)
+{
+  umpire::resource::MemoryResourceRegistry&
+    registry{ umpire::resource::MemoryResourceRegistry::getInstance() };
+  umpire::MemoryResourceTraits traits{ registry.getDefaultTraitsForResource(name) };
+  return traits;
+}
+
+void* find_pointer_from_name(Allocator allocator, const std::string name)
```

Comment:
Do you think we need to make this work for non-shared Allocators since they support named allocation

Reply:
I'm not sure I follow?

Reply:
This throws an error, but when calling `allocate(name, size)` on a non-shared allocator, the name is just ignored. I think it's a strange mismatch.

Reply:
But it's not.  Allocate(name,size) throws an error too: https://github.com/LLNL/Umpire/blob/cc10fea3cd7a9cc37ded9f58d8fb57aff2edaa57/src/umpire/strategy/AllocationStrategy.cpp#L31

Reply:
Ah okay. I don't think it used to, did it? We can resolve this then.

Reply:
Oh.  oops.  I intended for it to throw an error.